lib/concurrent/executor/executor.rb

```diff
@@ -65,22 +65,32 @@ def self.executor(executor_identifier)
       end
     end
 
-    # The policy defining how rejected tasks (tasks received once the
-    # queue size reaches the configured `max_queue`, or after the
-    # executor has shut down) are handled. Must be one of the values
-    # specified in `FALLBACK_POLICIES`.
-    attr_reader :fallback_policy
-
-    # def initialize(opts)
-    #   @auto_terminate = opts.fetch(:auto_terminate, true)
-    # end
-
-    def auto_terminate?
-      mutex.synchronize { ns_auto_terminate? }
+    # @!macro [attach] executor_method_post
+    #
+    #   Submit a task to the executor for asynchronous processing.
+    #
+    #   @param [Array] args zero or more arguments to be passed to the task
+    #
+    #   @yield the asynchronous task to perform
+    #
+    #   @return [Boolean] `true` if the task is queued, `false` if the executor
+    #     is not running
+    #
+    #   @raise [ArgumentError] if no task is given
+    def post(*args, &task)
```

Comment:
Any values needed within the task can be passed via `args`. Ruby has closures so any values needed within the task can be accessed via the enclosing context. Doing so, however, requires the runtime to retain the entire closure. This may be inefficient if the closure is large and only a few values are needed. Passing data as arguments instead will allow the runtime to release the closure.